integration/test/Test/Spar.hs

```diff
@@ -384,6 +385,37 @@ testSparIdPInitiatedLogin = do
     resp.status `shouldMatchInt` 200
     (cs resp.body) `shouldContain` "SAMLRequest"
 
+-- | UTF-8 chars (non-Latin-1) caused issues in XML parsing.
+testSparIdPInitiatedLoginUtf8Char :: (HasCallStack) => App ()
+testSparIdPInitiatedLoginUtf8Char = do
+  -- set up saml-not-scim team
+  (owner, tid, []) <- createTeam OwnDomain 1
+  void $ setTeamFeatureStatus owner tid "sso" "enabled"
+  (createIdpResp, (_idpmeta, privcreds)) <- registerTestIdPWithMetaWithPrivateCreds owner
+  assertSuccess createIdpResp
+
+  -- craft authnresp without req
+  idpValue :: A.Value <- createIdpResp.json
+  randomness <- randomId
+  let idp :: SAML.IdPConfig Value
+      idp = either error id $ A.parseEither (A.parseJSON @(SAML.IdPConfig A.Value)) idpValue
+      Right (subject :: SAML.NameID) =
+        SAML.mkNameID
+          ((SAML.mkUNameIDUnspecified . ST.pack) ("klăus-" ++ randomness))
+          Nothing
+          Nothing
+          Nothing
+  authnresp <- getAuthnResponseCustomNameID subject tid idp privcreds
+
+  -- send to finalize and check redirect response
+  finalizeSamlLogin OwnDomain tid authnresp `bindResponse` \resp -> do
+    -- the 303 is followed immediately, so the response is already coming from
+    -- /sso/initiate-login here.
+    resp.status `shouldMatchInt` 200
+    (cs resp.body) `shouldContain` "SAMLRequest"
+
+-- TODO: Can we assert that our special character klaus has been received correctly?
```

Comment:
yeah, this test isn't done yet.  the redirect is back to the idp with an authentication request, not into the app.  not what we want to test here i think, but it's probably a good test to keep anyway.